mkosi

```diff
@@ -739,19 +739,27 @@ def install_distribution(args, workspace, run_build_script):
     install[args.distribution](args, workspace, run_build_script)
 
 @complete_step('Resetting machine ID')
-def reset_machine_id(args, workspace):
+def reset_machine_id(args, workspace, run_build_script):
     """Make /etc/machine-id an empty file.
 
     This way, on the next boot is either initialized and commited (if /etc is
     writable) or the image runs with a transient machine ID, that changes on
     each boot (if the image is read-only).
     """
+
+    if run_build_script:
+        return
```

Comment:
This conflicts with using `complete_step` as a decorator. We shouldn't print anything if we skip the step.